File: solutions/3483-alternating-groups-ii/solution.py
class Solution:
    def numberOfAlternatingGroups(self, colors: List[int], k: int) -> int:
        alt_count = 0
        group_count = 0
        prev_color = colors[-1]
        l = len(colors)

        for i in range(0, l + k - 1):
            current_color = colors[i % l]

            if current_color != prev_color:
                alt_count += 1
            else:
                alt_count = 1

            if alt_count >= k:
                group_count += 1

            prev_color = current_color
            
        return group_count

File: solutions/3483-alternating-groups-ii/solution.js
/**
 * @param {number[]} colors
 * @param {number} k
 * @return {number}
 */
var numberOfAlternatingGroups = function(colors, k) {
    let alternating_count = 0
    let group_count = 0
    let prev_color = colors[colors.length - 1]

    for(let i = 0; i < colors.length + k - 1; i++){
      
        const index = i % colors.length;
        const current_color = colors[index]

        if(current_color === prev_color){
            alternating_count = 1;
        }else{
            alternating_count += 1;
        }

        if(alternating_count >= k){
            group_count += 1;
        }

        prev_color = current_color
    }

    return group_count;
};
